src/frequenz/sdk/actor/_power_managing/_bounds_methods.py

```diff
@@ -0,0 +1,143 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Methods for checking and clamping bounds and power values to exclusion bounds."""
+
+from ...timeseries import Bounds, Power
+
+
+def _check_exclusion_bounds_overlap(
+    lower_bound: Power,
+    upper_bound: Power,
+    exclusion_bounds: Bounds[Power] | None,
+) -> tuple[bool, bool]:
+    """Check if the given bounds overlap with the given exclusion bounds.
+
+    When only the upper bound overlaps with exclusion bounds, the usable range is
+    between the lower bound and the lower exclusion bound, like below.
+
+      ===lb+++++++ex----ub-------ex===
```

Comment:
To be honest, this sort of confuses me even more, because here you talk about usable range but the function talks about the exclusion zone.

Suggestion of a more explicit alternative:

> ```
>               lower                        upper
>                 .------ exclusion zone ------.
> -----|✔✔✔✔✔✔✔✔✔✔✔✔|xxxxxxxxxxxxxxx|----------|----
>      `-- usable --'-- exclusion --´
>      |                 overlap    |
>      |                            |
>    lower                        upper
>    bound                        bound
>                       (inside the exclusion zone)
> ```
> 
> Resulting in `(False, True)` because only the upper bound is inside the exclusion zone.

For me with this there is no need to do the other variants, seems clear enough with one example.

Reply:
That looks awesome, did you make that by hand?

Reply:
done

Reply:
> That looks awesome, did you make that by hand?

Yes :blush:

Reply:
I would love for GitHub/mkdocs to eventually support [aafigure](https://aafigure.readthedocs.io/en/latest/shortintro.html)